### PR TITLE
merge chart repo

### PR DESCRIPTION
**Pull request by acookin**
## Description

Result of merging `ambassador-chart.git` into `ambassador.git`, plus changes for CI to DTRT. Most of this is @acookin's handiwork; @kflynn just tossed the last few of commits on.

Big changes:

1. CI is in Circle, not Travis.
2. GitHub Actions are no longer used.
3. Releasing is done with `chart-vX.Y.Z` tags.

Review notes: 

*Don't* review all commits. The important ones:

- ef4d3e21 is @acookin's initial CI work, switching from Travis to Circle "for consistency and... travis CI makes [her] sad." 🙂 
- 07fc9e1a0 switches to using `chart-vX.Y.Z` tags to drive chart releases
- 92e827254 cleans up some junk with submodules and such
- 6ca338384 is there bceause we don't want to push the chart on Ambassador RCs (at least not right now)
- 3ee8730cd makes sure that we don't needlessly duplicate test setup when doing a chart release
- 6c719f2dd makes sure the `Chart.yaml` version and the tag match
- e169daeb5 makes sure we don't overwrite an existing chart version
 
The `CHANGELOG` for the chart is in `charts/ambassador/CHANGELOG.md`. There is no unified `CHANGELOG`. 

## Testing
@kflynn switched the CI machinery to look for `chart-test-vX.Y.Z` tags, and switched `push_chart.sh` to push to `test_$f` in the S3 bucket, and did a lot of tests by pushing `chart-test-` tags.

## Tasks That Must Be Done
- [x] Did you update CHANGELOG.md?
- [x] Is your change adequately tested?
- [ ] Did you update documentation?
- [ ] Were there special dev tricks you had to use to work on this code efficiently?
  + [ ] Did you add them to DEVELOPING.md?
- [x] This is not a build change, weirdly, just a CI change.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,5 +1,3 @@
-# Generated by `./generate`. DO NOT EDIT.
```
Comment (acookin) [inline .circleci/config.yml:1]: probably why oss generate failed :/

File: .circleci/config.yml.d/amb_util.yml [hunk 2]
```diff
@@ -132,3 +132,14 @@ commands:
...
           destination: test-logs
 
+  "amb-skip-if-bad-chart-version":
+    steps:
+      - run:
+          name: "Diff changes"
+          command: |
+            thisversion=$(grep version: charts/ambassador/Chart.yaml | awk ' { print $2 }')
+
+            if [[ "<< pipeline.git.tag >>" != "chart-test-v${thisversion}" ]]; then
```
Comment (acookin) [inline .circleci/config.yml.d/amb_util.yml:142]: i assume `-test` needs to be remove here?

File: CHANGELOG.md [hunk 3]
```diff
@@ -69,10 +69,11 @@ Please see the [Envoy documentation](https://www.envoyproxy.io/docs/envoy/latest
 
 ### Ambasssador API Gateway + Ambassador Edge Stack
 
+- Feature: Mapping configuration now supports setting `auth_context_extentions` that allows setting the `check_settings` field in the per route configuration supported by `ext_authz` http filter.
 - Bugfix: Make Knative paths match on prefix instead of the entire path to better align to the Knative specification ([#3224]).
+- Change: The Helm chart has been moved into this repo, in the `charts/ambassador` directory.
 
 [#3224]: https://github.com/datawire/ambassador/issues/3224
```
Comment (acookin) [inline CHANGELOG.md:76]: is this supposed to go with the knative entry in the changelog?

**Review by acookin — COMMENTED** [review]
woot! just one thing that i think you missed about the chart-test- tags

File: .circleci/config.yml (hunk 1 above)
Comment (kflynn) [reply .circleci/config.yml]: D'oh. Yeah. Thanks!

File: CHANGELOG.md (hunk 3 above)
Comment (kflynn) [reply CHANGELOG.md:76]: It's a separate line item -- not sure I follow what you're getting at here?

Comment (acookin) [reply CHANGELOG.md:76]: yeah... nevermind lol

File: .circleci/config.yml.d/amb_util.yml (hunk 2 above)
Comment (kflynn) [reply .circleci/config.yml.d/amb_util.yml:142]: Augh. Yes. I'll fix that.

**Review by acookin — COMMENTED** [review]
Woot lgtm

**Review by esmet — APPROVED** [review]

**Pull request merged**